--- /usr/share/X11/xkb/symbols/tr	2020-07-27 03:22:57.000000000 +0300
+++ ./linux/tr	2020-08-29 12:32:00.857249569 +0300
@@ -163,6 +163,71 @@
     name[Group1]="Turkish (Sun dead keys)";
 };
 
+partial alphanumeric_keys
+xkb_symbols "colemak" {
+
+  include "latin"
+  include "level3(ralt_switch)"
+
+  name[Group1]= "Turkish (Colemak)";
+
+  key <TLDE> {[ grave,       asciitilde  ]};
+  key <AE01> {[ 1,           exclam      ]};
+  key <AE02> {[ 2,           at          ]};
+  key <AE03> {[ 3,           numbersign  ]};
+  key <AE04> {[ 4,           dollar      ]};
+  key <AE05> {[ 5,           percent     ]};
+  key <AE06> {[ 6,           asciicircum ]};
+  key <AE07> {[ 7,           ampersand   ]};
+  key <AE08> {[ 8,           asterisk    ]};
+  key <AE09> {[ 9,           parenleft   ]};
+  key <AE10> {[ 0,           parenright  ]};
+  key <AE11> {[ minus,       underscore  ]};
+  key <AE12> {[ equal,       plus        ]};
+
+  key <AD01> {[ odiaeresis,  Odiaeresis  ]};
+  key <AD02> {[ scedilla,    Scedilla    ]};
+  key <AD03> {[ f,           F           ]};
+  key <AD04> {[ p,           P           ]};
+  key <AD05> {[ g,           G           ]};
+  key <AD06> {[ j,           J           ]};
+  key <AD07> {[ l,           L           ]};
+  key <AD08> {[ u,           U           ]};
+  key <AD09> {[ y,           Y           ]};
+  key <AD10> {[ gbreve,      Gbreve      ]};
+  key <AD11> {[ udiaeresis,  Udiaeresis  ]};
+  key <AD12> {[ backslash,   bar         ]};
+
+  key <AC01> {[ a,           A           ]};
+  key <AC02> {[ r,           R           ]};
+  key <AC03> {[ s,           S           ]};
+  key <AC04> {[ t,           T           ]};
+  key <AC05> {[ d,           D           ]};
+  key <AC06> {[ h,           H           ]};
+  key <AC07> {[ n,           N           ]};
+  key <AC08> {[ e,           E           ]};
+  key <AC09> {[ i,           Iabovedot   ]};
+  key <AC10> {[ o,           O           ]};
+  key <AC11> {[ idotless,    I           ]};
+  key <BKSL> {[ apostrophe,  quotedbl    ]};
+
+  key <LSGT> {[ minus,       underscore  ]};
+  key <AB01> {[ z,           Z           ]};
+  key <AB02> {[ ccedilla,    ccedilla    ]};
+  key <AB03> {[ c,           C           ]};
+  key <AB04> {[ v,           V           ]};
+  key <AB05> {[ b,           B           ]};
+  key <AB06> {[ k,           K           ]};
+  key <AB07> {[ m,           M           ]};
+  key <AB08> {[ comma,       less        ]};
+  key <AB09> {[ period,      greater     ]};
+  key <AB10> {[ slash,       question    ]};
+
+//key <CAPS> {[ BackSpace,   BackSpace    ]};
+  key <SPCE> {[ space,       nobreakspace ]};
+
+};
+
 // Kurdish Q layout
 // Erdal Ronahi <erdal.ronahi@gmail.com>, 2006
 // http://ferheng.org
